packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart

```diff
@@ -337,21 +356,29 @@ class EnrollVerbHandler extends AbstractVerbHandler {
     }
   }
 
-  Future<AtData> _fetchEnrollmentDataFromKeyStore(
-      String enrollmentKey, currentAtSign, String? enrollmentId) async {
-    AtData enrollData;
+  Future<AtData?> _fetchEnrollmentDataFromKeyStore(String enrollmentKey,
+      currentAtSign, String? enrollmentId, responseJson) async {
     // KeyStore.get will not return null. If the value is null, keyStore.get
-    // throws KeyNotFoundException.
-    // So, enrollData will NOT be null.
+    // throws KeyNotFoundException which will be rethrown as an EnrollmentException
+    AtData? enrollData;
     try {
       enrollData = await keyStore.get('$enrollmentKey$currentAtSign');
-    } on KeyNotFoundException {
+    } on KeyNotFoundException catch (e) {
+      responseJson['isError'] = 'true';
+      responseJson['errorCode'] = 'AT0028';
+      responseJson['errorMessage'] =
+          'enrollment_id: $enrollmentId is expired or invalid';
       throw AtEnrollmentException(
-          'enrollment id: $enrollmentId not found in keystore');
+          'enrollmentId: $enrollmentId is not an active key');
```

Comment:
can we have this message
enrollmentId: $enrollmentId is expired instead of is not an active key

Reply:
changed in [04c08a2](https://github.com/atsign-foundation/at_server/pull/1536/commits/04c08a2a6c3c5185aaccc7f02f1cdac7ea55d1eb)